app/policies/application_policy.rb

```diff
@@ -1,53 +1,72 @@
 class ApplicationPolicy
+  include ApplicationScope
+
   attr_reader :user, :record
 
   def initialize(user, record)
     @user = user
     @record = record
   end
 
-  def index?
-    false
+  delegate :superadmin?, to: :user
+  delegate :department_manager?, to: :user
+  delegate :social_worker?, to: :user
+  delegate :volunteer?, to: :user
+
+  def superadmin_or_social_worker?
+    superadmin? || social_worker?
   end
 
   def show?
     scope.where(id: record.id).exists?
   end
 
-  def create?
-    false
+  def scope
+    Pundit.policy_scope!(user, record.class)
   end
 
-  def new?
-    create?
+  def user_is_in?
+    user.present?
   end
 
-  def update?
+  def deny_all?
     false
   end
 
-  def edit?
-    update?
+  def allow_all?
+    true
   end
 
-  def destroy?
-    false
+  def user_owns_record?
+    record.user_id == user.id
   end
 
-  def scope
-    Pundit.policy_scope!(user, record.class)
+  def superadmin_or_record_owner?
+    superadmin? || user_owns_record?
   end
 
-  class Scope
-    attr_reader :user, :scope
+  def superadmin_or_user_in_records_related?
+    superadmin? || record.user_ids.include?(user.id)
+  end
 
-    def initialize(user, scope)
-      @user = user
-      @scope = scope
-    end
+  def superadmin_or_volunteers_record?
+    superadmin? || volunteer? && user_owns_record?
+  end
 
-    def resolve
-      scope
-    end
+  def superadmin_or_social_workers_record?
+    superadmin? || social_worker? && user_owns_record?
   end
+
+  def user_present_and_superadmin?
+    user && superadmin?
+  end
```

Comment:
Not sure if this is needed, if you do `delegate :superadmin, to: :user, allow_nil: true` in `ApplicationPolicy` you can just use `superadmin?` and remove this method.